### PR TITLE
Add unit test for DeviceIOControl

### PR DESCRIPTION
**Pull request by qmfrederik**
And `DISK_GEOMETRY`, `MEDIA_TYPE` struct & enum along the way.

### CODE REVIEW

File: src/Kernel32.Tests/storebanned/Kernel32Facts.cs [hunk 1]
```diff
@@ -1130,6 +1130,43 @@ unsafe uint GetProcessIdThreadProc(void* data)
...
+            flagsAndAttributes: 0,
+            SafeObjectHandle.Null))
+        {
+            Assert.False(device.IsInvalid);
+
+            DISK_GEOMETRY pdg = default;
+
+            Assert.True(DeviceIoControl(
+                device,
+                (int)IOCTL_DISK_GET_DRIVE_GEOMETRY,
```
Comment (qmfrederik) [inline src/Kernel32.Tests/storebanned/Kernel32Facts.cs:1154]: IOCTL codes really are `uint` values, not `int` values. They are 32-bit values which consist of various fields, see https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/defining-i-o-control-codes .

File: src/Kernel32.Tests/storebanned/Kernel32Facts.cs [hunk 2]
```diff
@@ -1130,6 +1130,43 @@ unsafe uint GetProcessIdThreadProc(void* data)
         GC.KeepAlive(threadStartRoutine); // Make sure that the delegate stays alive until the test is done
     }
 
+    [Fact]
+    public unsafe void DeviceIOControl_Works()
+    {
+        const uint IOCTL_DISK_GET_DRIVE_GEOMETRY = 0x070000;
```
Comment (qmfrederik) [inline src/Kernel32.Tests/storebanned/Kernel32Facts.cs:1136]: Common I/O codes are defined in https://docs.microsoft.com/en-us/windows/win32/api/winioctl/ , might make sense to add them to dotnet/pinvoke (as a separate effort)

File: src/Kernel32/storebanned/Kernel32+DISK_GEOMETRY.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+            /// <summary>
+            /// The number of sectors per track.
+            /// </summary>
+            public int SectorsPerTrack;
+
+            /// <summary>
+            /// The number of bytes per sector.
+            /// </summary>
+            public int BytesPerSector;
```
Comment (AArnott) [inline src/Kernel32/storebanned/Kernel32+DISK_GEOMETRY.cs]: nit: Given these values are unlikely to be handed to a .NET API, sticking with unsigned integers per the header file makes sense, IMO.

File: src/Kernel32.Tests/storebanned/Kernel32Facts.cs (hunk 1 above)
Comment (AArnott) [reply src/Kernel32.Tests/storebanned/Kernel32Facts.cs:1154]: That's unfortunate. I filed an issue to track eventually changing the signature to `uint` when we can take breaking changes.

**Review by AArnott — APPROVED** [review]
Thanks. We can merge as-is, but if you don't mind making the `int`->`uint` change I suggest, I think that would be slightly better.

File: src/Kernel32/storebanned/Kernel32+DISK_GEOMETRY.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        /// <summary>
+        /// Describes the geometry of disk devices and media.
+        /// </summary>
+        /// <seealso href="https://docs.microsoft.com/en-us/windows/win32/api/winioctl/ns-winioctl-disk_geometry"/>
+        public struct DISK_GEOMETRY
+        {
+            /// <summary>
+            /// The number of cylinders.
+            /// </summary>
+            public long Cylinders;
```
Comment (qmfrederik) [inline src/Kernel32/storebanned/Kernel32+DISK_GEOMETRY.cs:20]: Oddly enough, this is typed as [LARGE_INTEGER ](https://webcache.googleusercontent.com/search?q=cache:HVWWFBB64_kJ:https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-large_integer~r1+&cd=1&hl=en&ct=clnk&gl=be) which is signed.

**Comment by qmfrederik** [comment]
@AArnott Thanks, I made the `uint` -> `int`change, hope this PR is good to go.

**Pull request merged**